File: ts-playwright-cucumber/src/pages/checkoutOverviewPage.ts
import type { Page } from 'playwright';
import { BasePage } from './basePage.ts';
import { CheckoutCompletePage } from './checkoutCompletePage.ts';

const ITEM_NAMES = "[data-test='inventory-item-name']";
const TOTAL_LABEL = "[data-test='total-label']";
const FINISH_BUTTON = "[data-test='finish']";

export class CheckoutOverviewPage extends BasePage {
  constructor(page: Page) {
    super(page);
  }

  listItemNames(): Promise<string[]> {
    return this.page.locator(ITEM_NAMES).allTextContents();
  }

  getTotalLabel(): Promise<string> {
    return this.textOf(TOTAL_LABEL);
  }

  async finish(): Promise<CheckoutCompletePage> {
    await this.click(FINISH_BUTTON);
    return new CheckoutCompletePage(this.page);
  }
}

File: java-selenium-junit5-cucumber/src/test/java/com/saucedemo/uitests/pages/CheckoutOverviewPage.java
package com.saucedemo.uitests.pages;

import java.util.List;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public final class CheckoutOverviewPage extends BasePage {

    private static final By ITEM_NAMES = By.cssSelector("[data-test='inventory-item-name']");
    private static final By TOTAL_LABEL = By.cssSelector("[data-test='total-label']");
    private static final By FINISH_BUTTON = By.cssSelector("[data-test='finish']");

    public CheckoutOverviewPage(WebDriver driver) {
        super(driver);
    }

    public List<String> listItemNames() {
        return driver.findElements(ITEM_NAMES).stream().map(WebElement::getText).toList();
    }

    public String totalLabel() {
        return textOf(TOTAL_LABEL);
    }

    public CheckoutCompletePage finish() {
        click(FINISH_BUTTON);
        return new CheckoutCompletePage(driver);
    }
}
